mantidimaging/gui/windows/live_viewer/model.py

```diff
@@ -12,12 +12,100 @@
 LOG = getLogger(__name__)
 
 
+class Image_Data:
+    """
+    Image Data Class to store represent image data.
+
+    ...
+
+    Attributes
+    ----------
+    image_path : str
+        path to image file
+    image_name : str
+        name of image file
+    image_size : int
+        size of image file
+    image_modified_time : int
+        last modified time of image file
+    """
+
+    def __init__(self, image_path: Path):
+        """
+        Constructor for Image_Data class.
+
+        Parameters
+        ----------
+        image_path : str
+            path to image file
+        """
+        self.image_path: Path = image_path
+        self.image_name: str = image_path.name
+        self._image_size: int
+        self._image_modified_time: float
+        self._set_file_info()
```

Comment:
Can I suggest that only the path and result of `stat()` are stored as attributes, and then properties are used to get the other values.

```python
    self.image_path = image_path
    self.stat = image_path.stat()
```

Then you can drop the setters. And have

```python
    @property
    def image_size(self):
        """Return the image size"""
        return self.stat.st_size
```
This would simplify this class, and mean that `stat()` only gets called once. And it saves the setters having a different input than the value.

Maybe let the exception bubble up and catch it in _get_image_files() if needed.

Also, you should not need to type annotate `self.image_path` because it is set from an annotated argument to `__init__`.

Reply:
I mostly agree with this, I will still need to wrap stat in a try catch as stat may not exist when trying to set initially if the file is removed or in a partial state during initialisation.